pkg/collector/runner/runner.go

```diff
@@ -398,18 +398,28 @@ func addWorkStats(c check.Check, execTime time.Duration, err error, warnings []e
 }
 
 func expCheckStats() interface{} {
-	checkStats.M.RLock()
-	defer checkStats.M.RUnlock()
-
-	return checkStats.Stats
+	return GetCheckStats()
 }
 
 // GetCheckStats returns the check stats map
 func GetCheckStats() map[string]map[check.ID]*check.Stats {
 	checkStats.M.RLock()
 	defer checkStats.M.RUnlock()
 
-	return checkStats.Stats
+	// because the returned maps will be used after the lock is released, and
+	// thus when they might be further modified, we must clone them here.  The
+	// map values (`check.Stats`) are threadsafe and need not be cloned.
+
+	cloned1 := make(map[string]map[check.ID]*check.Stats)
```

Comment:
Nit: Can we rename these `cloned*` vars to indicate what they are cloning? I think having `cloned1` and `cloned2` contain different structures in close proximity can be a bit confusing to quickly read/maintain.

Reply:
I'm not sure what the names are for the two levels of this map?  id-prefix and id?

Reply:
..I'll just use `innerCloned`, `innerK`, etc.